modules/core/src/main/java/org/apache/synapse/aspects/newstatistics/StatisticsEntry.java

```diff
@@ -0,0 +1,510 @@
+	 */
+	private int clonedMsgCount = -1;
+
+	/**
+	 * This overloaded constructor will create the root statistic log og the Statistic Entry
+	 * according to given parameters. Statistic Event for creating statistic entry can be either
+	 * PROXY, API or SEQUENCE.
+	 *
+	 * @param componentId     componentId of the statistic reporting element
+	 * @param componentType   component Type of the statistic reporting element
+	 * @param msgId           msgId of the statistics reporting message context
+	 * @param parentId        parentId of the reporting statistic event
+	 * @param startTime       starting time of the statistics reporting event
+	 * @param isResponse      is message context belong to an response
+	 * @param localMemberHost localMemberHost in the cluster
+	 * @param localMemberPort localMemberPort in the cluster
+	 */
+	public StatisticsEntry(String componentId, ComponentType componentType, int msgId, String parentId, long startTime,
+	                       boolean isResponse, String localMemberHost, String localMemberPort) {
+		this.localMemberHost = localMemberHost;
+		this.localMemberPort = localMemberPort;
+		StatisticsLog statisticsLog = new StatisticsLog(componentId, componentType, msgId, -1, -1, parentId, startTime);
+		statisticsLog.setIsResponse(isResponse);
+		messageFlowLogs.add(statisticsLog);
+		openLogs.addFirst(messageFlowLogs.size() - 1);
+		if (log.isDebugEnabled()) {
+			log.debug("Created statistic Entry [Start|RootElement]|[ElementId|" + componentId +
+			          "]|[MsgId|" + msgId + "]");
+		}
+	}
+
+	/**
+	 * Create statistics log at the start of a statistic reporting element
+	 *
+	 * @param componentId   componentId of the statistic reporting element
+	 * @param componentType component Type of the statistic reporting element
+	 * @param msgId         msgId of the statistics reporting message context
+	 * @param parentId      parentId of the reporting statistic event
+	 * @param startTime     starting time of the statistics reporting event
+	 * @param isResponse    is message context belong to an response
+	 */
+	public synchronized void createLog(String componentId, ComponentType componentType, int msgId, String parentId,
+	                                   long startTime, boolean isResponse) {
+		if (openLogs.isEmpty()) {
+			StatisticsLog statisticsLog = new StatisticsLog(componentId, componentType, msgId, -1, -1, "", startTime);
+			messageFlowLogs.add(statisticsLog);
+			openLogs.addFirst(messageFlowLogs.size() - 1);
+			if (log.isDebugEnabled()) {
+				log.debug("Starting statistic log at root level [ElementId|" + componentId +
+				          "]|[MsgId|" + msgId + "]");
+			}
+		} else {
+			if (openLogs.getFirst() == 0) {
+				if (messageFlowLogs.get(0).getComponentId().equals(componentId)) {
+					if (log.isDebugEnabled()) {
+						log.debug("Statistics event is ignored as it is a duplicate of root " + "element");
+					}
+					return;
+				}
+			}
+			if (parentId.equals("")) {
```

Comment:
If Java 7/8 isn't being used, you can possibly keep the semantics of the comparison same with a small modification to swap the two values compared within the if clause, in order to avoid possible NPEs when the parentId is null.